Improve unpacking of solutions

hannesbraun opened the pull request:
In the case that students wrap their solutions with unnecessary folders, the multi-checker will now try to determine the actual directory that contains the solution.

Fixes #2

hannesbraun committed:
Improve unpacking of solutions

In the case that students wrap their solutions with unnecessary folders,
the multi-checker will now try to determine the actual directory that
contains the solution.

skogsbaer commented on multi-checker/script/utils.py:
```diff
@@ -82,3 +82,31 @@ def isDebug():
+    if stopCondition(root):
+        return root
+    for x in ls(root):
+        result = findDirMatchingAux(x, stopCondition)
```
only descend recursively if root is a wrapper dir. For example, you do not want to descend in the following case:

root/A/src
root/B/src

hannesbraun replied:
This should be fixed with 8140d9dcbca827ba3c592f320c5ead875b561e00.

skogsbaer commented on multi-checker/script/utils.py:
```diff
@@ -82,3 +82,31 @@ def isDebug():
+        return False
+    return not isDir(pathnames[0])
+
+def findDirMatching(root: str, stopCondition: Callable) -> str:
```
Short documentation

hannesbraun replied:
Done

skogsbaer reviewed (changes requested):
I think a couple of test cases would be good.

hannesbraun committed:
Improve solution unpacking

The function got renamed to findSolutionDir as it is specifically made
to find a solution (not a general directory matching just a condition).
Some short documentation was added to the function as well.
Also, it does now stop descending if either the stop condition is true
or the directory is not a wrapper directory.

hannesbraun committed:
Add some tests to check if findSolutionDir works
